python_modules/libraries/dagster-k8s/dagster_k8s/container_context.py

```diff
@@ -370,56 +408,6 @@ def _get_used_k8s_config_fields(self) -> Mapping[str, Mapping[str, Set[str]]]:
                 {field_key for field_key in fields}
             )
 
-        if self.image_pull_policy:
-            # image_pull_policy is equivalent to container_config.image_pull_policy
-            used_fields["container_config"].add("image_pull_policy")
-
-        if self.image_pull_secrets:
-            # image_pull_secrets is equivalent to pod_spec_config.image_pull_secrets
-            used_fields["pod_spec_config"].add("image_pull_secrets")
-
-        if self.service_account_name:
-            # service_account_name is equivalent to pod_spec_config.service_account_name
-            used_fields["pod_spec_config"].add("service_account_name")
-
-        if self.env_config_maps or self.env_secrets:
-            # env_config_maps and env_secrets end up on env_from on the container
-            used_fields["container_config"].add("env_from")
-
-        if self.env_vars:
-            # env_vars end up in the "env" field on the container
-            used_fields["container_config"].add("env")
-
-        if self.env:
-            # env is equivalent to container_config.env
-            used_fields["container_config"].add("env")
-
-        if self.volume_mounts:
-            # volume_mounts is equivalent to container_config.volume_mounts
-            used_fields["container_config"].add("volume_mounts")
-
-        if self.volumes:
-            # volumes is equivalent to pod_spec_config.volumes
-            used_fields["pod_spec_config"].add("volumes")
-
-        if self.labels:
-            # labels end up on both pod_template_spec_metadata.labels nad
-            # job_metadata.labels since they apply to both the pod and job
-            used_fields["pod_template_spec_metadata"].add("labels")
-            used_fields["job_metadata"].add("labels")
-
-        if self.resources:
-            # resources is equivalent to container_config.resources
-            used_fields["container_config"].add("resources")
-
-        if self.scheduler_name:
-            # scheduler_name is equivalent to pod_spec_config.scheduler_name
-            used_fields["pod_spec_config"].add("scheduler_name")
-
-        if self.security_context:
-            # security_context is equivalent to container_config.security_context
-            used_fields["container_config"].add("security_context")
-
```

Comment:
this parts gets a lot cleaner - instead of having to special-case each top-level field and see where it will be inserted in the k8s manifest, we can just look at the k8s manifest